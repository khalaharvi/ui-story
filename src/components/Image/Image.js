import React from 'react';
import PropTypes from 'prop-types';

// TO DO: IMGIX + srcset/sizes

const Image = ({ alt, className, src }) => (
  <img src={src} alt={alt} className={className} />
);

Image.propTypes = {
  className: PropTypes.string,
  alt: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired
};

export default Image;
