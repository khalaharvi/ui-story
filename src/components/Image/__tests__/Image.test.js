import React from 'react';
import { shallow } from 'enzyme';

import Image from '../Image';

describe('<Image />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Image src="https://placehold.it/120x20" alt="Placeholder" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with a className prop', () => {
    const wrapper = shallow(<Image src="https://placehold.it/120x20" alt="Placeholder" className="image" />);
    expect(wrapper).toMatchSnapshot();
  });
});
