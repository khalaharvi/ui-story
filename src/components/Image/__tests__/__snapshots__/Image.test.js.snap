// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> renders correctly 1`] = `
<img
  alt="Placeholder"
  src="https://placehold.it/120x20"
/>
`;

exports[`<Image /> renders correctly with a className prop 1`] = `
<img
  alt="Placeholder"
  className="image"
  src="https://placehold.it/120x20"
/>
`;
